source/brailleDisplayDrivers/nattiqbraille.py

```diff
@@ -0,0 +1,140 @@
+# brailleDisplayDrivers/nattiqbraille.py
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2011-2020
+
+
+import serial
+import braille
+import inputCore
+from logHandler import log
+import hwIo
+
+BAUD_RATE = 10000000
+TIMEOUT = 0.3
+INIT_TAG = "0"
+
+
+class BrailleDisplayDriver(braille.BrailleDisplayDriver):
+	name = "nattiqbraille"
+	# Translators: Names of braille displays
+	description = _("Nattiq nBraille")
+	isThreadSafe = True
+
+	@classmethod
+	def getManualPorts(cls):
+		return braille.getSerialPorts()
+
+	def __init__(self, port="auto"):
+		super(BrailleDisplayDriver, self).__init__()
+		self._serial = None
+		for portType, portId, port, portInfo in self._getTryPorts(port):
+			log.debug("Checking port %s for a Nattiq nBraille", port)
+			try:
+				self._serial = hwIo.Serial(
+					port, baudrate=BAUD_RATE, timeout=TIMEOUT, writeTimeout=TIMEOUT,
+					parity=serial.PARITY_NONE, onReceive=self._onReceive
+				)
+			except EnvironmentError:
+				log.debugWarning("", exc_info=True)
+				continue
+			# Check for cell information
+			if self._describe():
+				log.debug("Nattiq nBraille found on %s with %d cells", port, self.numCells)
+				break
+			else:
+				self._serial.close()
+		else:
+			raise RuntimeError("Can't find a Nattiq nBraille device (port = %s)" % port)
+
+	def terminate(self):
+		try:
+			super(BrailleDisplayDriver, self).terminate()
+		finally:
+			self._serial.write("reset".encode())
+			self._serial.close()
+			self._serial = None
+
+	def _describe(self):
+		self.numCells = 0
+		log.debug("Writing reset tag")
+		self._serial.write("reset".encode())
+		self._serial.waitForRead(3)

```

Comment:
This is quite a long time.

```suggestion
		self._serial.waitForRead(self.timeout * 10)

```

Reply:
I edited this out, and it seems to works fine, so I will keep it.
The reason why I put 3 is that I was not sure how long the time is needed for the system unit to do the "reset", but it seems fast enough that the change to "self.timeout * 10" does not create a problem.